pkg/controllers/user/monitoring/operatorHandler.go

```diff
@@ -23,6 +23,11 @@ func (h *operatorHandler) syncCluster(key string, obj *mgmtv3.Cluster) (runtime.
 	if obj == nil || obj.DeletionTimestamp != nil || obj.Name != h.clusterName {
 		return obj, nil
 	}
+
+	if !mgmtv3.ClusterConditionAgentDeployed.IsTrue(obj) {
```

Comment:
why are you adding this agent deployed check? How does it help address the bug?

Reply:
@cjellick , This error is caused by two problems: 
- projectClient cannot list all project
- the error reason & msg cannot be cleaned

Agent deployed checking can ensure two things:
1. the CRDs of Prometheus Operator are installed (For all types of clusters)
2. the connection from the server to the local cluster

Add agent deployed checking can help us: 
- make sure never install Operator chart before the CRDs not ready.
- for local cluster, make sure never do the `sync*` logics before the agent takes over the connections for the local cluster, which may cause `projectClient cannot list all project`.

cc @loganhz